builtin/bytes.mbt

```diff
@@ -14,7 +14,26 @@
 
 /// Create a new String from byte sequence.
 pub fn to_string(self : Bytes) -> String {
-  self.sub_string(0, self.length())
+  let sb = StringBuilder::new(size_hint=self.length())
+  fn to_hex_digit(i : Int) -> Char {
+    if i < 10 {
+      Char::from_int('0'.to_int() + i)
+    } else {
+      Char::from_int('a'.to_int() + (i - 10))
+    }
+  }
+
```

Comment:
can we print ascii as is?
e.g,    b"a\0b\0"

Reply:
It's kind hard to find the ascii inside a bunch of `\00`. Also encounter: `Fatal error: exception File "inlined_snapshot/_build/Core_format.ml", line 200, characters 8-14: Assertion failed`. Investigating

Reply:
> can we print ascii as is? e.g, b"a\0b\0"

displaying some byte as ASCII may not be a good idea. `Bytes` are not always converted from a String, they may come from data of an image or some pieces of memory. ASCII is meaningless in such cases.

Reply:
Second this as UTF-16 encoded string has lots of `\x00` and printing them is very ugly.